src/fs/protocol/public/index.ts

```diff
@@ -86,12 +88,14 @@ export const getValue = async (
   linksOrCID: SimpleLinks | CID,
   name: string,
 ): Promise<unknown> => {
-  if (isString(linksOrCID)) {
-    const links = await basic.getSimpleLinks(linksOrCID)
+  const cid = CID.asCID(linksOrCID)
+
+  if (cid) {
+    const links = await basic.getSimpleLinks(cid)
     return getValueFromLinks(links, name)
   }
 
-  return getValueFromLinks(linksOrCID, name)
+  return getValueFromLinks(linksOrCID as SimpleLinks, name)
```

Comment:
Goddamnit. This whole `CID.asCID` pattern that PL has come up with breaks typescript type inference.
We *could* write a `(obj: unkown): obj is CID => notNull(CID.asCID(obj))` function, but I think that defeats the purpose, because you should still run `CID.asCID` on whatever you got & then typescript will complain that that returns something that can be null.
Hm. Let me know if I'm missing something we could do to help this situation.

Reply:
Yeah I'm not sure what Typescript is doing here.

This does not work:
<img width="994" alt="Screenshot 2022-01-24 at 23 02 47" src="https://user-images.githubusercontent.com/296665/150872499-57b8dbca-6e65-48a2-9373-af5780bda489.png">
_If I remove the `&& cid` here, it gives me a "could null" error in the first `if` branch._

This does:
<img width="508" alt="Screenshot 2022-01-24 at 23 04 44" src="https://user-images.githubusercontent.com/296665/150872733-bded3dc4-e61f-467e-8038-92ee695665f2.png">
_Should I do this instead?_

Reply:
I've given this another thought. Let's make the `CID.asCID` calls mostly happen on the surface-level API boundary. This function is pretty internal. So let's change it to something like this:
```ts

  if (check.isCID(linksOrCID)) {
    const links = await basic.getSimpleLinks(linksOrCID)
    return getValueFromLinks(links, name)
  }
```